### PR TITLE
Add publisher entity 455

### LINKED ISSUE #455: Add publisher entity
**Describe the solution you'd like**
Add a publisher entity. There should be a one-to-many relationship between a `Publisher` and a `Book` (where a `Publisher` is the owner).

For now, the publisher entity should contain only one field (besides an ID that is inherited from `BaseEntity`): a name.

There only needs to be a respective repository and service.

**Additional context**
This is a backend-only change.

Update: there should be a many-to-many relationship

### PR DESCRIPTION
**Pull request by abhity16**
## Summary of change

Added Publisher Entity,Service, Repository. Modified Other classes (like Book etc)  accordingly. Added the Test cases for this. 

## Related issue

Closes #455 

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-jcijyenp-JiKFGBv62FIPoFnvOW6Ubg)

### CODE REVIEW

**Comment by abhity16** [comment]
Hi @knjk04 
Is there anything more to do in this issue. I'm just wondering why is this PR marked as draft

**Comment by knjk04** [comment]
Hi @abhity16, could you please fill in the PR template? #538 is an example of where the PR template was filled in

**Comment by abhity16** [comment]
Hi @knjk04 
I've updated the comment. Please review the PR. 
Thanks,
Abhinav

**Comment by abhity16** [comment]
@knjk04 I'll be away for next whole week. So please wait for me to work upon any comments on PR (unless urgent)

**Comment by knjk04** [comment]
Hi Abhinav,

That's fine, thanks for the update. We'll review this PR this weekend.

File: src/test/java/com/karankumar/bookproject/backend/service/PublisherServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        publisherService.save(publisher);
+        Long publisherId = publisher.getId();
+
+        // when
+        publisherService.delete(publisher);
+
+        // then
+        assertThatThrownBy(() -> publisherService.findById(publisherId)).isInstanceOf(
+                JpaObjectRetrievalFailureException.class);
+    }
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/service/PublisherServiceTest.java:103]: Is it not possible to delete a publisher?

Comment (abhity16) [reply src/test/java/com/karankumar/bookproject/backend/service/PublisherServiceTest.java:103]: Yes, we can. 

What I'm trying here to delete the publisher and checking the existance of that publisher after that. It should throw an error .

Comment (knjk04) [reply src/test/java/com/karankumar/bookproject/backend/service/PublisherServiceTest.java:103]: Ah, my bad. I should have looked closer at the assertion. Thanks for clarifying

**Review by knjk04 — APPROVED** [review]
Thanks for working on this!

**Review by knjk04 — CHANGES_REQUESTED** [review]
Could you please address the review feedback?

**Comment by abhity16** [comment]
@knjk04 I've changed the return type.

**Review by knjk04 — APPROVED** [review]
Thanks for making the change and thanks again for working on this!

**Pull request merged**